src/main/java/com/oltpbenchmark/benchmarks/tpch/TPCHLoader.java

```diff
@@ -335,49 +373,90 @@ public void beforeLoad() {
-                        switch (type) {
-                            case DOUBLE:
-                                prepStmt.setDouble(idx + 1, (Double) elems.get(idx));
-                                break;
-                            case LONG:
-                                prepStmt.setLong(idx + 1, (Long) elems.get(idx));
-                                break;
-                            case STRING:
-                                prepStmt.setString(idx + 1, (String) elems.get(idx));
-                                break;
-                            case DATE:
-                                prepStmt.setDate(idx + 1, (Date) elems.get(idx));
-                                break;
-                            default:
-                                throw new RuntimeException("Unrecognized type for prepared statement");
-                        }
-                    }
-
+                    
+                    addElems(prepStmt, elems, types);
                     ++recordsRead;
+                    currentBatch.add(elems);
                     prepStmt.addBatch();
                     if ((recordsRead % workConf.getBatchSize()) == 0) {
-
-                        LOG.debug("writing batch {} for table {}", recordsRead, tableName);
-
-                        prepStmt.executeBatch();
-                        prepStmt.clearBatch();
+                        int retryCount = 0;
+                        while (retryCount < maxRetryCount){
+                            try {
+                                LOG.debug("writing batch {} for table {}", recordsRead, tableName);
+                                prepStmt.executeBatch();
+                                prepStmt.clearBatch();
+                                currentBatch.clear();
+                                break;
+                            }catch (Exception e) {
+                                LOG.error(e.getMessage(), e);
+                                
+                                retryCount++;
+                                if (retryCount == maxRetryCount){
+                                    currentBatch.clear();
+                                    LOG.error("Failed batch {} for table {}", recordsRead, tableName);
+                                }
+                                else{
+                                    if (!prepStmt.getConnection().isValid(1000)) {
+                                        prepStmt.close();
+                                        conn = this.benchmark.makeConnection();
+                                        prepStmt = loader.getPreparedStatement(conn);
+                                        for (List<Object> retryElems : currentBatch){
+                                            addElems(prepStmt, retryElems, types);
+                                            prepStmt.addBatch();
+                                        }
+                                    }
```

Comment:
What is this logic? Why are you making a new connection? Why not have the retry logic outside of this function so that you can catch, rebuild the batch, and then try again?